### PR TITLE
[Devtools/radio-group] asset

### PR DESCRIPTION
**Pull request by lexfm**
### Change Type (required)

Implementation of `@devtools-ui/radio-group` asset.

**Implemented (not limited to) the following:**

docs/storybook/.storybook/preview.ts
.bazelignore
pnpm-workspace.yaml
plugin/src/plugins/AssetsRegistryPlugin.tsx
plugin/src/plugins/TransformPlugin.ts
plugin/src/index.ts
DSL
React component
Transformer
README
DSL tests
Storybook stories
Docs
 

**Extended scope:**

- Designed and implemented slots for radio options for `@devtools-ui/radio-group` to utilize.
- Linked nav items for MDX docs site from assets created prior gen:asset improvements.
- Improved naming convention for docs site side navigation for generated assets.

- [ ] `patch`
- [x] `minor`
- [ ] `major`

### CODE REVIEW

**Comment by KetanReddy** [comment]
Is there any reason we're exposing radio item as a separate asset?

**Comment by lexfm** [comment]
> Is there any reason we're exposing radio item as a separate asset?

It was decided to follow the same asset convention for design and reusability.

File: docs/site/pages/assets/radio-group.mdx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+// and use it to define your Player-UI content:
+myFlow = {
+    id: 'my_flow',
+    views: [
+        <>
+            <RadioGroup />
+        </>
+    ]
+}
```
Comment (rafbcampos) [inline docs/site/pages/assets/radio-group.mdx]: Let's also add the example of RadioGroup.Values usage:
```suggestion
    views: [
      <RadioGroup binding={b`my_binding`}>
        <RadioGroup.Label>RadioGroup Label</RadioGroup.Label>
        <RadioGroup.Values>
          <RadioGroup.Values.Value value="opt1">
            <RadioGroup.Values.Value.Label>
              Option 1
            </RadioGroup.Values.Value.Label>
          </RadioGroup.Values.Value>
          <RadioGroup.Values.Value value="opt2">
            <RadioGroup.Values.Value.Label>
              Option 2
            </RadioGroup.Values.Value.Label>
          </RadioGroup.Values.Value>
        </RadioGroup.Values>
      </RadioGroup>
    ]
}
```

File: .bazelignore [hunk 2]
```diff
@@ -31,6 +31,8 @@ table/node_modules
 copy-to-clipboard/node_modules
 toggle/node_modules
 code-editor/node_modules
+radio-group/node_modules
+radio-item/node_modules
```
Comment (rafbcampos) [inline .bazelignore:35]: We can remove the radio-item:
```suggestion
```

File: docs/storybook/src/assets/RadioGroup.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+It provides a `RadioGroup` component that can present a way to input one of multiple choices.
+
+## RadioGroup DSL Properties
+
+| Property | Type | Required | Description |
+| --- | --- | --- | --- |
+| `binding` | `string` | false | The location in the data-model to store the data |
+| `label` | `AssetWrapper<TextAsset> \| string` | false | A text asset for the radio group label |
+| `values` | `Array<AssetWrapper<RadioGroup.Value.Values>>` | false | An array of options containing value and label) |
```
Comment (rafbcampos) [inline docs/storybook/src/assets/RadioGroup.mdx:18]:
```suggestion
| `values` | `Array<AssetWrapper<RadioGroup.Value.Values>>` | false | An array of options containing value and label |
```

**Review by rafbcampos — APPROVED** [review]

File: docs/storybook/src/assets/RadioGroup.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+const meta: Meta<typeof RadioGroup> = {
+  title: "Devtools UI Assets/RadioGroup",
+  component: RadioGroup,
+};
+
+export default meta;
+
+export const Basic = createDSLStory(
+  () => import("../flows/radio-group/basic?raw")
```
[the remark this answers is not in the document]
Comment (rafbcampos) [reply docs/storybook/src/assets/RadioGroup.stories.tsx:13]: https://vitejs.dev/guide/assets#importing-asset-as-string

**Pull request merged**